build/bazel/remote/execution/v2/remote_execution.proto

```diff
@@ -855,6 +974,35 @@ message ExecutedActionMetadata {
 
   // When the worker finished uploading action outputs.
   google.protobuf.Timestamp output_upload_completed_timestamp = 10;
+
+  // The number of dynamic directory information requests made to the client.
+  // Used with
+  // [ExecuteWithDynamicRetrieval][build.bazel.remote.execution.v2.Execution.ExecuteWithDynamicRetrieval].
+  int32 num_dynamic_directory_requests = 11;
+
+  // The total time spent by the worker waiting for dynamic directory
+  // information from the client. Used with
+  // [ExecuteWithDynamicRetrieval][build.bazel.remote.execution.v2.Execution.ExecuteWithDynamicRetrieval].
+  google.protobuf.Duration dynamic_directory_requests_time = 12;
+
+  // The number of dynamic file hash requests made to the client. Used with
+  // [ExecuteWithDynamicRetrieval][build.bazel.remote.execution.v2.Execution.ExecuteWithDynamicRetrieval].
+  int32 num_dynamic_file_hash_requests = 13;
+
+  // The total time spent by the worker waiting for dynamic file hash
+  // information from the client. Used with
+  // [ExecuteWithDynamicRetrieval][build.bazel.remote.execution.v2.Execution.ExecuteWithDynamicRetrieval].
+  google.protobuf.Duration dynamic_file_hash_requests_time = 14;
+
+  // The number of dynamic file content requests made to the client when the
+  // worker could not find a dynamic file hash in the CAS. Used with
+  // [ExecuteWithDynamicRetrieval][build.bazel.remote.execution.v2.Execution.ExecuteWithDynamicRetrieval].
+  int32 num_dynamic_file_content_requests = 15;
+
+  // The total time spent by the worker waiting for dynamic file content
+  // information from the client. Used with
+  // [ExecuteWithDynamicRetrieval][build.bazel.remote.execution.v2.Execution.ExecuteWithDynamicRetrieval].
+  google.protobuf.Duration dynamic_file_content_requests_time = 16;
```

Comment:
Should this be put in an inner `message`, i.e., `message DynamicExecutionMetadata` ?

Reply:
Good idea. Done.

---
In reply to: [459002553](https://github.com/erikma/remote-apis/pull/1#discussion_r459002553) [](ancestors = 459002553)